proposals/compatibility-mode.md

```diff
@@ -209,6 +209,13 @@ In addition to the new limits, the existing `maxStorageBuffersPerShaderStage` an
 
 **Justification**: OpenGL ES 3.1 allows `MAX_VERTEX_SHADER_STORAGE_BLOCKS` and `MAX_VERTEX_IMAGE_UNIFORMS` to be zero, and there are a significant number of devices in the field with that value.
 
+## 20. Limit the number of texture+sampler combinations in a pipeline.
+
+If the number of texture+sampler combinations used a in single stage in a pipeline exceeds
```

Comment:
Should we clarify that combinations includes textures without samplers as well?

Reply:
Yikes. That's an interesting issue.  A texture used without a sampler counts as 1 but only if it's not used somewhere else? Or do we keep it simple? In other words, if you have

```wgsl
@group(0) @binding(0) var t: texture_2d<f32>;
@group(0) @binding(1) var s: sampler;

   x = textureLoad(t);
   y = textureSampler(t, s);
```
   
That's seems like it only counts as 1 combo

I'm not sure the easiest way to state that.

> The number of texture+sampler combinations is counted by first counting the number of texture+sampler combinations, then adding 1 for every texture used without a sampler that's not already counted.

or something like that?

Or do we count the example above as 2 combos? `[(t+s), (t)]`

Reply:
Checking dawn with the code above it only created 1 `uniform sampler2D`

Reply:
I think it should be 1 combo. Probably can be stated as essentially `max(used without sampler ? 1 : 0, number of texture sampler combos)`

Reply:
updated

Reply:
Is this also complicated by external texture? It seems like it is like maybe it should be `min(maxSampledTexturesPerShaderStage - numExternalTexturesUsedInStage * 4, maxSamplerPerShaderStage - numExternalTexturesUsageInStage * 2)`

Reply:
I'm not sure `max(used without sampler ? 1 : 0, number of texture sampler combos)` is correct. I reverted it back to how it was before for now 

`max(used without sampler ? 1 : 0, number of texture sampler combos)` would make sense if it said something like, 

> The number of texture+sampler combinations is counted as follows:
>
> ```
> sum = 0;
> for each texture used in entry point
>    sum += max(texture+sampler combos for texture, texture used without sampler ? 1 :  0)

or something like that

Reply:
I think that is what I had in mind, I didn't sketch it very carefully. Though just realizing, the 0/1 is accounted for by the bind group layouts, and it doesn't matter if it's actually used with `textureLoad`? So:
```
for each stage of the pipeline:
  sum = 0
  for each texture binding in the pipeline layout which is visible to that stage:
    sum += max(1, number of texture sampler combos for that texture binding)
```

Good point about external texture, that definitely needs to be explicit about how it interacts with this limit. The internal resources of an external texture are explained in the note here: https://gpuweb.github.io/gpuweb/#gpuexternaltexture

So the accounting (https://gpuweb.github.io/gpuweb/#exceeds-the-binding-slot-limits) would add something like this, via the logic above:
```
    sum += 1 // for LUT texture + LUT sampler
    sum += 3 * max(1, number of external_texture sampler combos) // for Y+U+V
```

Reply:
ok, added, ptal